src/analyzer/protocol/ssl/tls-handshake-protocol.pac

```diff
@@ -378,33 +349,57 @@ type ServerKeyExchange(rec: HandshakeRecord) = case $context.connection.chosen_c
 	TLS_DH_ANON_WITH_CAMELLIA_128_GCM_SHA256,
 	TLS_DH_ANON_WITH_CAMELLIA_256_GCM_SHA384
 	# DH non-anon suites do not send a ServerKeyExchange
-		-> dh_server_key_exchange : DhServerKeyExchange(rec);
+		-> dh_anon_server_key_exchange : DhAnonServerKeyExchange(rec);
 
 	default
 		-> key : bytestring &restofdata &transient;
 };
 
-# For the moment, we really only are interested in the curve name. If it
-# is not set (if the server sends explicit parameters), we do not bother.
-# We also do not parse the actual signature data following the named curve.
-type EcServerKeyExchange(rec: HandshakeRecord) = record {
+# Parse an ECDHE ServerKeyExchange message, which contains a signature over the
+# parameters. Parsing explicit curve parameters from the server is not
+# currently supported.
+type EcdheServerKeyExchange(rec: HandshakeRecord) = record {
 	curve_type: uint8;
 	curve: uint16; # only if curve_type = 3 (NAMED_CURVE)
+	point_length: uint8;
```

Comment:
I am going to fix this - but just to leave a note - doing things like this will lead to exceptions in case someone is using non-named-curve. Binpac will still try to parse the rest of the handshake and error out.